src/librustc_mir/hair/cx/mod.rs

```diff
@@ -105,11 +105,7 @@ impl<'a,'tcx:'a> Cx<'a, 'tcx> {
 
     pub fn needs_drop(&mut self, ty: Ty<'tcx>, span: Span) -> bool {
         if self.infcx.type_moves_by_default(ty, span) {
```

Comment:
Is this first `if` redundant now?  Maybe the whole method body can be simplified  to a single function call?